File: 3044-minimum-operations-to-collect-elements/3044-minimum-operations-to-collect-elements.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def min_operations(nums, k)
    target = k * (k+1) / 2
    seen = Set.new
    curr = 0
    nums.reverse.each_with_index do |n,i|
        next if seen.include?(n) || !(1..k).cover?(n)
        curr += n
        seen.add(n)
        return i + 1 if curr == target
    end
    -1
end

File: 3044-minimum-operations-to-collect-elements/3044-minimum-operations-to-collect-elements.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        target = k * (k+1) // 2
        curr = 0
        seen = set()
        nums.reverse()
        for i, n in enumerate(nums):
            if n in seen or not 1 <= n <= k:
                continue
            curr += n
            seen.add(n)
            if curr == target:
                return i + 1
        return -1

